mcp/streamable.go

```diff
@@ -44,6 +47,10 @@ type StreamableHTTPHandler struct {
 type StreamableHTTPOptions struct {
 	// TODO: support configurable session ID generation (?)
 	// TODO: support session retention (?)
+
+	// TransportOptions sets the streamable server transport options to use when
+	// establishing a new session.
+	TransportOptions func(*http.Request) *StreamableServerTransportOptions
```

Comment:
Why is this a function on the request? (Use cases?)

Reply:
Let's think for a moment if there's a possible use. You can remove for now to make progress.

Reply:
I unexported.